src/lib/testing/equivalent.ts

```diff
@@ -235,16 +244,63 @@ let bool: ProvableSpec<boolean, Bool> = {
   back: (x) => x.toBoolean(),
   provable: Bool,
 };
-let boolean: Spec<boolean, boolean> = {
-  rng: Random.boolean,
-  there: id,
-  back: id,
-};
+let boolean: Spec<boolean, boolean> = fromRandom(Random.boolean);
 
 function fieldWithRng(rng: Random<bigint>): Spec<bigint, Field> {
   return { ...field, rng };
 }
 
+// spec combinators
+
+function array<T, S>(
+  spec: ProvableSpec<T, S>,
+  n: number
+): ProvableSpec<T[], S[]>;
+function array<T, S>(
+  spec: Spec<T, S>,
+  n: Random<number> | number
+): Spec<T[], S[]>;
```

Comment:
Are these function declarations?
Why is the second declaration needed (it's the same type as the implementation right below)?

Reply:
This is how you declare if there are two different function signatures. The third one is not exposed, it's the "internal" signature used in writing the function. So if I remove the second one, consumers only see the first one